packages/vitest/src/node/stdin.ts

```diff
@@ -36,6 +39,37 @@ ${keys
   )
 }
 
+function* traverseFilteredTestNames(stack: string[], filter: RegExp, t: Task): Generator<FilterType> {
+  if (isAtomTest(t)) {
+    if (t.name.match(filter)) {
+      const displayName = `${stack.join(' > ')} > ${t.name}`
+      yield { name: t.name, toString: () => displayName }
+    }
+  }
+  else {
+    for (const task of t.tasks) {
+      yield * traverseFilteredTestNames([...stack, task.name], filter, task)
```

Comment:
add `parentName: string` instead recreating the array

Reply:
The latest commit fixes duplicate test names - check the screenshot, `t1` appears twice